File: src/weaponsprocurement/gui/WimGuiInputResult.java
package weaponsprocurement.gui;

final class WimGuiInputResult {
    private static final WimGuiInputResult NONE = new WimGuiInputResult(false, 0);
    private static final WimGuiInputResult CLOSE = new WimGuiInputResult(true, 0);

    private final boolean closeRequested;
    private final int scrollDelta;

    private WimGuiInputResult(boolean closeRequested, int scrollDelta) {
        this.closeRequested = closeRequested;
        this.scrollDelta = scrollDelta;
    }

    static WimGuiInputResult none() {
        return NONE;
    }

    static WimGuiInputResult closeRequested() {
        return CLOSE;
    }

    static WimGuiInputResult scroll(int delta) {
        return delta == 0 ? NONE : new WimGuiInputResult(false, delta);
    }

    boolean isCloseRequested() {
        return closeRequested;
    }

    boolean hasScrollDelta() {
        return scrollDelta != 0;
    }

    int getScrollDelta() {
        return scrollDelta;
    }
}

File: src/main/kotlin/weaponsprocurement/gui/WimGuiInputResult.kt
package weaponsprocurement.gui

class WimGuiInputResult private constructor(
    private val closeRequested: Boolean,
    private val scrollDelta: Int,
) {
    fun isCloseRequested(): Boolean = closeRequested

    fun hasScrollDelta(): Boolean = scrollDelta != 0

    fun getScrollDelta(): Int = scrollDelta

    companion object {
        private val NONE = WimGuiInputResult(false, 0)
        private val CLOSE = WimGuiInputResult(true, 0)

        @JvmStatic
        fun none(): WimGuiInputResult = NONE

        @JvmStatic
        fun closeRequested(): WimGuiInputResult = CLOSE

        @JvmStatic
        fun scroll(delta: Int): WimGuiInputResult = if (delta == 0) NONE else WimGuiInputResult(false, delta)
    }
}
